### PR TITLE
Display the location and fix tests.

### LINKED ISSUE #89: Display the location of the user on the map.
Display the location of the user on the map.
Should request permission from the user to access the location.
Should be able to center the map on the location.

### LINKED ISSUE #127: Fix `MapDrawerAndroidTest`.
Should instantiate the provider directly instead of relying on injeciton from hilt.
Clean up unnecessary tests, and remove exposed `mapView` fields from the `OsmdroidMapViewProvider`.

### LINKED ISSUE #129: Default view of the map on the EPFL
Set the map default zoom and position centered on the EPFL campus

### LINKED ISSUE #178: Fix Map not showing after navigating to event list and back.

### PR DESCRIPTION
**Pull request by alejandrocalles**
Closes #89.
Closes #127.
Closes #129.
Closes #178.

### CODE REVIEW

**Comment by alejandrocalles** [comment]
Closes #127.

**Comment by alejandrocalles** [comment]
This PR solves a bunch of different but related issues.

The main one is #89, but solving #127 was needed to get the test coverage and #129 was too short to get a PR of its own.

**Comment by alejandrocalles** [comment]
I tried to fix #163 to no avail. For the moment, i have set the initial bearing to 0.0 and completely disabled rotation.

**Review by Greenade — CHANGES_REQUESTED** [review]
The idea is good and the code looks great, the position blue thing looks good as well! There are some problems though : 
When we give the permission to display the position, it doesn't display it until we swap pages and come back, so I think you should refresh the map when receiving the permission to display the position. 
Another thing is that for #163 you made it worse x) at least on the emulator when we barely spin the map it crashes now, It doesn't do that on my phone though (doesn't even crash when clicking the compass). 
One last thing is I think the map should be centered arount the position when given the permission.

**Review by Greenade — DISMISSED** [review]

**Comment by Greenade** [comment]
I removed my requested changes as I cannot review your branch :/ please ask somebody to verify it works though 🥲

File: app/src/main/java/com/github/swent/echo/compose/map/Map.kt [hunk 1]
```diff
@@ -22,11 +34,45 @@ fun <T : View> EchoAndroidView(
...
 ) {
-    AndroidView(
-        modifier = modifier.testTag("mapAndroidView"),
-        factory = factory,
-        update = { update(it, events, callback) }
+    val permissions = arrayOf(
+        Manifest.permission.ACCESS_COARSE_LOCATION,
+        Manifest.permission.ACCESS_FINE_LOCATION
     )
+    fun permissionsDenied(context: Context) = permissions.any {
```
Comment (unglazedstamp) [inline app/src/main/java/com/github/swent/echo/compose/map/Map.kt]: why is this function declared inside the EchoAndroidView function ?

**Review by unglazedstamp — CHANGES_REQUESTED** [review]
Good code ! I found problems by running the app:
1. If I refuse location permission it ask it again when I click on an event pin, it only does it the first time but it shouldn't at all
2. Some pins in the background becomes black when I click on one of them
3. If I allow approximate location, go to the list and come back to the map, it asks again if I want precise location
The second issue is not necessarly related to this PR and none of them are not blocking the app, so if you have the time to solve 1. and 3. it would be nice, otherwise we can merge this PR and solve them later.

File: app/src/main/java/com/github/swent/echo/compose/map/Map.kt (hunk 1 above)
Comment (alejandrocalles) [reply app/src/main/java/com/github/swent/echo/compose/map/Map.kt]: Good catch! I've moved it out.

**Review by unglazedstamp — APPROVED** [review]

**Comment by unglazedstamp** [comment]
All the problem are solved when I run the app on a real phone instead of the emulator. Thanks a lot for all this work !

**Review by unglazedstamp — APPROVED** [review]

**Pull request merged**